CHANGELOG.md

```diff
@@ -1,5 +1,29 @@
+## 4.0.0 (November 9, 2020)
+
+BREAKING CHANGES:
+
+- `pattern` replaces `branch` in changes to `github_branch_protection` introduced in `v3.1.0` ([#566](https://github.com/terraform-providers/terraform-provider-github/issues/566))
+- `dismissal_restrictions` documented for `github_branch_protection` ([#569](https://github.com/terraform-providers/terraform-provider-github/pull/569))
+- project license changes from MPL2 to MIT ([#591](https://github.com/terraform-providers/terraform-provider-github/pull/591))
+
+BUG FIXES:
+
+- `repository` re-instated alongside `repository_id` for `github_branch_protection` ([#566](https://github.com/terraform-providers/terraform-provider-github/issues/566))
```

Comment:
This is not yet implemented, but will take a pass at it today as requested [here](https://github.com/terraform-providers/terraform-provider-github/issues/566#issuecomment-713709445).

/cc @onwsk8r

Reply:
```suggestion
- `repository_id` for `github_branch_protection` accepts repository name as well as node ID([#593](https://github.com/terraform-providers/terraform-provider-github/issues/593))
```